examples/train_model.py

```diff
@@ -157,15 +172,21 @@ def validate(self):
         if opt.get('model_file') and opt.get('save_after_valid'):
             print("[ saving model checkpoint: " + opt['model_file'] + ".checkpoint ]")
             self.agent.save(opt['model_file'] + '.checkpoint')
+            self.world.save_agents()
+            if self.best_valid is not None:
+                print("[ saving best valid metric: " + opt['model_file'] + ".checkpoint.best_valid ]")
+                save_best_valid(opt['model_file'] + '.checkpoint', self.best_valid)
```

Comment:
I think maybe better to not save here. when loading a checkpoint, you should probably `cp/mv model.checkpoint model` and then run again, or else the next run will be saving to model.checkpoint.checkpoint and so on... not good. this also makes it look like maybe saved model at model.checkpoint got the model.checkpoint.best_valid score, which isn't true.

(so let's remove these four lines. also just realized that line 175 is unfortunately saving over the best valid model.)